scipy/sparse/tests/test_construct.py

```diff
@@ -776,24 +776,22 @@ def test_rand(self):
             assert_(np.all(np.less_equal(0, x.data)))
             assert_(np.all(np.less_equal(x.data, 1)))
 
-    def test_randn(self):
+    @pytest.mark.parametrize("rng", [None, 4321, np.random.RandomState(4321),
+                                     np.random.default_rng(4321)])
+    def test_randn(self, rng):
         # Simple distributional checks for sparse.randn.
         # Statistically, some of these should be negative
         # and some should be greater than 1.
-        random_states = [None, 4321, np.random.RandomState()]
-        try:
-            gen = np.random.default_rng()
-            random_states.append(gen)
-        except AttributeError:
-            pass
+        x = _sprandn(10, 20, density=0.5, dtype=np.float64, random_state=rng)
+        assert_(np.any(np.less(x.data, 0)))
+        assert_(np.any(np.less(1, x.data)))
+
+        if isinstance(rng, np.random.RandomState):  # not accepted by `rng`
```

Comment:
Looks like we have lost the test that passing in `RandomState()` instance works. I prefer not to have this change potentially make a few users reluctant to migrate to `sparray` because it doesn't handle `RandomState`. In my view we should either find a way to allow RandomState instances to work, or make this shift to "rng" happen with `spmatrix` tools as well as `sparray` tools (e.g. `rand`, `randn`, `sprand` and maybe others I'm not recalling). If the library as a whole is going to no longer work with RandomState then let's put these changes into the spmatrix tools too.

Reply:
I could add this to show that it works during the transition:
```suggestion
        x = _sprandn_array(10, 20, density=0.5, dtype=np.float64, random_state=rng)
        assert_(np.any(np.less(x.data, 0)))
        assert_(np.any(np.less(1, x.data)))
        
        if isinstance(rng, np.random.RandomState):  # not accepted by `rng`
```
Basically, we test `_sprandn_array` twice, once with the old `random_state` keyword, which accepts `RandomState`, and once with the new `rng` keyword, which doesn't.

I didn't think this was necessary because the these tests only ensure that the argument is accepted - not that the `RandomState` or other object does anything - and the decorator has been tested to pass a `random_state` argument straight through, unmodified. But easy enough to add it now, if desired!